backend/src/main/java/com/fake_orgasm/usersmanagement/libs/btree/BTree.java

```diff
@@ -0,0 +1,377 @@
+            x.setChild(j + 1, x.getChild(j));
+        }
+        x.setChild(pos + 1, z);
+        for (int j = x.getSize() - 1; j >= pos; j--) {
+            x.setKey(j + 1, x.getKey(j));
+        }
+        x.setKey(pos, y.getKey(order - 1));
+        x.setSize(x.increaseSize());
+    }
+
+
+    /**
+     * Inserts a key into the B-tree. The B-tree structure might be modified to accommodate the new key.
+     *
+     * @param key The key to insert.
+     * @throws IllegalArgumentException If the provided key is null.
+     */
+    public void insert(final T key) {
+        if (key == null) throw new IllegalArgumentException("Key cannot be null");
+        Node<T> r = root;
+        if (r.getSize() == 2 * order - 1) {
+            Node<T> node = new Node<>(order);
+            root = node;
+            node.setLeaf(false);
+            node.setSize(0);
+            node.setChild(0, r);
+            split(node, 0, r);
+            insertIntoSubtree(node, key);
+        } else {
+            insertIntoSubtree(r, key);
+        }
+    }
+
+    /**
+     * Inserts a new key into the B-tree starting from the specified subtree node.
+     *
+     * @param key  The key to be inserted.
+     * @param node The starting node of the subtree.
+     */
+    private void insertIntoSubtree(Node<T> node, T key) {
+
+        if (node.isLeaf()) {
+            int i;
+            for (i = node.getSize() - 1; i >= 0 && key.compareTo(node.getKey(i)) < 0; i--) {
+                node.setKey(i + 1, node.getKey(i));
+            }
+
+            node.setKey(i + 1, key);
+            node.setSize(node.increaseSize());
+        } else {
+            int i = node.getSize() - 1;
+            while (i >= 0 && key.compareTo(node.getKey(i)) < 0) {
+                i--;
+            }
+            i++;
+            Node<T> tmp = node.getChild(i);
+            if (tmp.getSize() == 2 * order - 1) {
+                split(node, i, tmp);
+                if (key.compareTo(node.getKey(i)) > 0) {
+                    i++;
+                }
+            }
+            insertIntoSubtree(node.getChild(i), key);
+        }
+
+    }
+
+
+    /**
+     * Prints the entire B-tree structure by recursively traversing and printing each node's keys.
+     * The tree is printed in an indented format to show hierarchy.
+     */
+    public void printTree() {
+        this.printTree(root);
+    }
+
+    public void printTree(Node<T> x) {
+        if(x == null)return;
+        for (int i = 0; i < x.getSize(); i++) {
+            System.out.print(x.getKey(i)+ " ");
+        }
+        if (!x.isLeaf()) {
+            for (int i = 0; i < x.getSize() + 1; i++) {
+                printTree(x.getChild(i));
+            }
+        }
+    }
+
+
+    /**
+     * Removes a key from the B-tree starting from the given node.
+     *
+     * @param x   The current node being considered.
+     * @param key The key to remove.
+     */
+
+    private void remove(Node<T> x, T key) {
```

Comment:
Maybe you could try to divide this method a little more to make it easier to read or change in the future.